aiogram/dispatcher/fsm/storage/mongo.py

```diff
@@ -0,0 +1,230 @@
+import asyncio
+from contextlib import asynccontextmanager
+from datetime import datetime, timedelta
+from typing import Any, AsyncGenerator, Dict, Optional
+
+from aiogram import Bot
+from aiogram.dispatcher.fsm.state import State
+from aiogram.dispatcher.fsm.storage.base import DEFAULT_DESTINY, BaseStorage, StateType, StorageKey
+from motor.motor_asyncio import AsyncIOMotorClient, AsyncIOMotorDatabase  # type: ignore
+from pymongo.errors import InvalidOperation, DuplicateKeyError
+from pymongo.results import UpdateResult
+
+STATE = "aiogram_state"
+DATA = "aiogram_data"
+LOCK = "aiogram_lock"
+COLLECTIONS = (STATE, DATA, LOCK)
+
+DEFAULT_MONGO_LOCK = 60
+
+
+class MongoStorage(BaseStorage):
+    """
+    Mongo storage required :code:`motor` package installed (:code:`pip install motor`)
+    """
+
+    def __init__(
+            self,
+            mongo: AsyncIOMotorClient,
+            db: AsyncIOMotorDatabase,
+            with_bot_id: bool = True,
+            with_destiny: bool = True,
+            state_ttl: Optional[int] = None,
+            data_ttl: Optional[int] = None,
+            lock_ttl: Optional[int] = None,
+    ) -> None:
+        self._mongo = mongo
+        self._db = db
+        self._with_bot_id = with_bot_id
+        self._with_destiny = with_destiny
+        self.state_ttl = state_ttl
+        self.data_ttl = data_ttl
+        self.lock_ttl = lock_ttl
+
+    @classmethod
+    async def from_url(
+            cls,
+            url: str,
+            db_name: str = "aiogram_fsm",
```

Comment:
Hi! Sorry for the intrusion, but if turning over a new leaf for v3.0, then it would be great if workarounds for using the URI-provided db_name would be unnecessary (see my PR for v2: https://github.com/aiogram/aiogram/pull/844).

I'd suggest something along the lines of:
```
...
db_name: Optional[str] = None
...
if db_name is None:
    db = mongo.get_default_database()
else:
    db = mongo.get_database(self.default_db_name)
...
```
That would also allow one to subclass and redefine the default database (the `default_db_name` class attribute) if needed.

And also kwargs would be great for custom SSL options and stuff